tests/Unit/Helpers/Domain/DomainTestHelpers.cpp

```diff
@@ -504,6 +505,49 @@ void test_initial_domain(const Domain<VolumeDim>& domain,
   domain::test_refinement_levels_of_neighbors<1>(elements);
 }
 
+// Given a Block, tests that the determinant of the Jacobian is positive
+// at all corners of that Block.
+template <size_t VolumeDim>
+void test_det_jac_positive(
+    const std::vector<Block<VolumeDim>>& blocks, double time,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time) noexcept {
+  std::array<tnsr::I<double, VolumeDim, Frame::Logical>, two_to_the(VolumeDim)>
+      corner_points{};
+  size_t indx = 0;
+  // Need only lower_xi() and upper_xi() directions to get all the corners.
+  for (auto direction :
+       {Direction<VolumeDim>::lower_xi(), Direction<VolumeDim>::upper_xi()}) {
+    for (FaceCornerIterator<VolumeDim> fci(direction); fci; ++fci) {
+      gsl::at(corner_points, indx++) = fci();
+    }
+  }
```

Comment:
The loops would be simpler (one less loop), but VolumeCornerIterator returns a `std::array<double,VolumeDim>` instead of a `tnsr::I<double,VolumeDim,Frame::Logical>`, so using VolumeCornerIterator would mean that I need to convert the array to a tensor.

So for now I haven't changed this. Please let me know if I should.

Reply:
FYI, there's also `Domain/Structure/Hypercube.hpp`. I think `VolumeCornerIterator` and `FaceCornerIterator` could probably be expressed as a `HypercubeElementsIterator`. Just letting you know it exists :)

Reply:
I think the conversion can be done inline with a cast or a call to a Tensor constructor.  If not, as-is is fine.  No preference on my part between VCI and Hypercube.